econml/tests/test_model_selection.py

```diff
@@ -0,0 +1,135 @@
+        return Y, T, X, W
+
+    def test_poly(self):
+        # tests that we can recover the right degree of polynomial features
+        # implicitly also tests ability to handle pipelines
+        # since 'poly' uses pipelines containing PolynomialFeatures
+        n = 5000
+        X = np.random.normal(size=(n, 2))
+        W = np.random.normal(size=(n, 3))
+
+        for true_d in range(1, 4):
+            with self.subTest(true_d=true_d):
+                pf = PolynomialFeatures(degree=true_d)
+
+                fts = pf.fit_transform(np.hstack([X, W]))
+                k = fts.shape[1]
+                m = X.shape[1] + W.shape[1]
+
+                alpha_x = np.random.normal(size=(X.shape[1],))
+                alpha_1 = np.random.normal(size=())
+                beta = np.random.normal(size=(k,))
+                gamma = np.random.normal(size=(k,))
+
+                # generate larger coefficients in a set of high degree features,
+                # weighted towards higher degree features
+                ft_inds_beta = np.random.choice(k, size=m, replace=False, p=np.arange(k) / np.sum(np.arange(k)))
+                ft_inds_gamma = np.random.choice(k, size=m, replace=False, p=np.arange(k) / np.sum(np.arange(k)))
+
+                beta[ft_inds_beta] = 10 * np.random.normal(1, size=(m,))
+                gamma[ft_inds_gamma] = 10 * np.random.normal(1, size=(m,))
+
+                t = np.random.normal(size=(n,)) + fts @ beta + np.random.normal(scale=0.5, size=(n,))
+                y = np.random.normal(size=(n,)) + t * (alpha_1 + X @ alpha_x) + fts @ gamma
+
+                # just test a polynomial T model, since for Y the correct degree also depends on
+                # the interation of T and X
+                mdl = LinearDML(model_t='poly',
+                                model_y=LinearRegression()).fit(y, t, X=X, W=W)
+                for t in mdl.models_t[0]:
+                    self.assertEqual(t[0].degree, true_d)
+
+    def test_all_strings(self):
+        for discrete_treatment in [True, False]:
+            Y, T, X, W = self._simple_dgp(500, 2, 3, discrete_treatment)
+            for model_t in ['auto', 'linear', 'poly', 'forest', 'gbf', 'nnet', 'automl']:
+                with self.subTest(model_t=model_t, discrete_treatment=discrete_treatment):
+                    mdl = LinearDML(model_t=model_t,
+                                    discrete_treatment=discrete_treatment,
+                                    model_y=LinearRegression())
+                    mdl.fit(Y, T, X=X, W=W)
+
+            model_t = 'some_random_string'
+            with self.subTest(model_t=model_t, discrete_treatment=True):
+                mdl = LinearDML(model_t=model_t,
+                                discrete_treatment=discrete_treatment,
+                                model_y=LinearRegression())
+                with self.assertRaises(ValueError):
+                    mdl.fit(Y, T, X=X, W=W)
+
+    def test_list_selection(self):
```

Comment:
Is it worth checking other possible permutations like [str, str], [str, SklearnCVselector], [fixedSelector, str].

Reply:
Wouldn't hurt, but I believe the current tests already cover all of the relevant code paths.